test/board/devices/input/switches.test.ts

```diff
@@ -0,0 +1,256 @@
+  switches.toggle(20)
+  switches.toggle(22)
+  switches.toggle(24)
+  switches.toggle(26)
+  switches.toggle(28)
+  switches.toggle(30)
+})
+
+test('test that switches are set to readonly', () => {
+  switches.writeByte(block1, byte_0000_0000)
+  switches.writeByte(block2, byte_0000_0000)
+  switches.writeByte(block3, byte_0000_0000)
+  switches.writeByte(block4, byte_0000_0000)
+
+  expect(switches.readByte(block1)).toEqual(byte_0101_0101)
+  expect(switches.readByte(block2)).toEqual(byte_0101_0101)
+  expect(switches.readByte(block3)).toEqual(byte_0101_0101)
+  expect(switches.readByte(block4)).toEqual(byte_0101_0101)
+})
+
+describe('test isOn() function', () => {
+  test('test passing invalid positions to isOn() function', () => {
+    expect(() => switches.isOn(32)).toThrow(
+      new Error('Position 32 does not exist.')
+    )
+    expect(() => switches.isOn(-2)).toThrow(
+      new Error('Position -2 does not exist.')
+    )
+  })
+  test('test passing valid positions to isOn() function', () => {
+    expect(switches.isOn(0)).toBe(true)
+    expect(switches.isOn(1)).toBe(false)
+    expect(switches.isOn(2)).toBe(true)
+    expect(switches.isOn(3)).toBe(false)
+    expect(switches.isOn(4)).toBe(true)
+    expect(switches.isOn(5)).toBe(false)
+    expect(switches.isOn(6)).toBe(true)
+    expect(switches.isOn(7)).toBe(false)
+
+    expect(switches.isOn(8)).toBe(true)
+    expect(switches.isOn(9)).toBe(false)
+    expect(switches.isOn(10)).toBe(true)
+    expect(switches.isOn(11)).toBe(false)
+    expect(switches.isOn(12)).toBe(true)
+    expect(switches.isOn(13)).toBe(false)
+    expect(switches.isOn(14)).toBe(true)
+    expect(switches.isOn(15)).toBe(false)
+
+    expect(switches.isOn(16)).toBe(true)
+    expect(switches.isOn(17)).toBe(false)
+    expect(switches.isOn(18)).toBe(true)
+    expect(switches.isOn(19)).toBe(false)
+    expect(switches.isOn(20)).toBe(true)
+    expect(switches.isOn(21)).toBe(false)
+    expect(switches.isOn(22)).toBe(true)
+    expect(switches.isOn(23)).toBe(false)
+
+    expect(switches.isOn(24)).toBe(true)
+    expect(switches.isOn(25)).toBe(false)
+    expect(switches.isOn(26)).toBe(true)
+    expect(switches.isOn(27)).toBe(false)
+    expect(switches.isOn(28)).toBe(true)
+    expect(switches.isOn(29)).toBe(false)
+    expect(switches.isOn(30)).toBe(true)
+    expect(switches.isOn(31)).toBe(false)
+  })
+})
+
+describe('test set() function', () => {
+  test('test passing invalid positions to set() function', () => {
+    expect(() => switches.set(32, true)).toThrow(
+      new Error('Position 32 does not exist.')
+    )
+    expect(() => switches.set(-2, false)).toThrow(
+      new Error('Position -2 does not exist.')
+    )
+  })
+  test('test passing valid positions to set() function', () => {
+    switches.set(1, true)
+    switches.set(3, true)
+    switches.set(5, true)
+    switches.set(7, true)
+
+    switches.set(8, false)
+    switches.set(10, false)
+    switches.set(12, false)
```

Comment:
perhaps set some values which are already set to true and vice versa. Or does this make no sense since the dip switch on click always changes the value?

Reply:
Created a new test for this.